From 4b24af9e9f3524550a4b8f5666c10315a45dfe05 Mon Sep 17 00:00:00 2001
From: William Pitcock <nenolod@dereferenced.org>
Date: Thu, 2 Nov 2017 04:51:51 +0000
Subject: [PATCH] abuild: add support for provider_priority

provider_priority is a number which determines what priority a package should be
given when solving a dependency graph using a provides entry instead of a direct
package, in the event of conflicts.
---
 abuild.in | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/abuild.in b/abuild.in
index 3f083b2..87aeb66 100644
--- a/abuild.in
+++ b/abuild.in
@@ -139,6 +139,12 @@ default_sanitycheck() {
 		&& ! echo $replaces_priority | egrep -q '^[0-9]+$'; then
 		die "replaces_priority must be a number"
 	fi
+
+	if [ -n "$provider_priority" ] \
+		&& ! echo $provider_priority | egrep -q '^[0-9]+$'; then
+		die "provider_priority must be a number"
+	fi
+
 	# check so no package names starts with -
 	for i in $pkgname $subpackages; do
 		case $i in
@@ -927,6 +933,10 @@ prepare_metafiles() {
 		echo "replaces_priority = $replaces_priority" >> "$pkginfo"
 	fi
 
+	if [ -n "$provider_priority" ]; then
+		echo "provider_priority = $provider_priority" >> "$pkginfo"
+	fi
+
 	echo "license = $license" >> "$pkginfo"
 	for i in $replaces; do
 		echo "replaces = $i" >> "$pkginfo"
